File: projects/grind-75/python/tree/bt_level_order_traversal.py
from collections import deque


class Solution:
    @classmethod
    def levelOrder(self, root):

        if not root:
            return []

        res = []
        q = deque([root])

        while q:
            res.append([node.val for node in q])
            for _ in range(len(q)):
                node = q.popleft()
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)

        return res

File: projects/grind-75/typescript/tree/bt-level-order-traversal.ts
import { TreeNode } from "./common"

export default function levelOrder(root: TreeNode | null): number[][] {
    
    if(!root) return []
    const q = [root]
    const res = []

    while (q.length){
        res.push(q.map(node=>node.val))
        const currLenQ = q.length
        for(let i=0; i< currLenQ; i++){
            const node = q.shift()!
            if(node.left){
                q.push(node.left)
            }
            if(node.right){
                q.push(node.right)
            }
        }
    }

    return res
};